### PR TITLE
Added celery task logger to the security library

### PR DESCRIPTION
**Pull request by matllubos**

### CODE REVIEW

File: security/models.py [hunk 1]
```diff
@@ -142,32 +145,26 @@ def prepare_from_request(self, request):
         )
 
 
-class LoggedRequest(models.Model):
+class LoggedRequestStatus(ChoiceEnum):
```
Comment (matllubos) [inline security/models.py]: new enums

File: security/models.py [hunk 2]
```diff
@@ -142,32 +145,26 @@ def prepare_from_request(self, request):
...
-    STATUS_CHOICES = (
-        (INCOMPLETE, _('Incomplete')),
-        (INFO, _('Info')),
-        (WARNING, _('Warning')),
-        (ERROR, _('Error')),
-        (DEBUG, _('Debug')),
-        (CRITICAL, _('Critical')),
-    )
+
+class LoggedRequest(SmartModel):
```
Comment (matllubos) [inline security/models.py]: model inherits SmartModel now

File: security/models.py [hunk 3]
```diff
@@ -281,7 +273,10 @@ def update_from_response(self, response):
     class Meta:
         verbose_name = _('input logged request')
         verbose_name_plural = _('input logged requests')
-        ordering = ('-request_timestamp',)
+        ordering = ('-created_at',)
+
+    class SmartMeta:
+        is_cleaned_pre_save = False
```
Comment (matllubos) [inline security/models.py]: pre save clean is turned off due to performance issues

File: security/models.py [hunk 4]
```diff
@@ -338,4 +336,55 @@ class CommandLog(models.Model):
...
+class CeleryTaskManager(models.Manager):
+
+    def filter_stale(self, **kwargs):
+        return self.filter(
+            state__in={CeleryTaskLogState.WAITING, CeleryTaskLogState.ACTIVE, CeleryTaskLogState.RETRIED},
+            changed_at__lt=timezone.now() - timedelta(minutes=settings.CELERY_STALE_TASK_TIME_LIMIT_MINUTES)
+        )
+
+
+class CeleryTaskLog(SmartModel):
```
Comment (matllubos) [inline security/models.py]: new model

File: example/settings/celery.py [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import os
+import sys
+
+from celery import Celery
+
+from django.conf import settings
+
+app = Celery('mallpay')
```
Comment (Formulka) [inline example/settings/celery.py]: that's an oddly specific example name :trollface:

File: security/tasks.py [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+            pass
+        self._call_callback('failure', task_id, args, kwargs)
+
+    def apply_async_on_commit(self, args=None, kwargs=None, **options):
+        task = CeleryTaskLog.objects.create(name=self.name, state=CeleryTaskLogState.WAITING,
+                                            queue_name=options.get('queue', settings.CELERY_DEFAULT_QUEUE))
+        args = (task.pk,) + tuple(args or ())
+        self.on_apply(task.pk, args, kwargs)
+        if sys.argv[1:2] == ['test']:
+            super().apply_async(args=args, kwargs=kwargs, task_id=task.pk, **options)
```
Comment (Formulka) [inline security/tasks.py]: everywhere else you are casting the task.pk to str, shouldn't you do the same here?

File: security/tasks.py [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+        else:
+            super_inst = super()
+            transaction.on_commit(
+                lambda: super_inst.apply_async(args=args, kwargs=kwargs, task_id=str(task.pk), **options)
+            )
+
+    def apply_async(self, args=None, kwargs=None, **options):
```
Comment (Formulka) [inline security/tasks.py]: apply_async_on_commit and apply_async are so similar, isn't there a way to cut something out into a helper or is this for the sake of clarity?

File: security/models.py (hunk 1 above)
Comment (Formulka) [reply security/models.py]: how about putting them in an enums.py file?

File: example/settings/celery.py (hunk 5 above)
Comment (matllubos) [reply example/settings/celery.py]: shiit

File: security/models.py (hunk 1 above)
Comment (matllubos) [reply security/models.py]: I don't see reason, there are not probability of circle imports

File: security/tasks.py (hunk 6 above)
Comment (matllubos) [reply security/tasks.py]: yes thanks

File: security/tasks.py (hunk 7 above)
Comment (matllubos) [reply security/tasks.py]: yes we can

**Pull request merged**